lettuce/boundary.py

```diff
@@ -93,3 +108,14 @@ def __call__(self, f):
              - (torch.norm(u_w,dim=0) / self.lattice.cs) ** 2)
         )
         return f
+
+    def make_no_stream_mask(self, f_shape):
+        no_stream_mask = torch.zeros(size=f_shape, dtype=torch.bool, device=self.lattice.device)
+        no_stream_mask[[np.array(self.lattice.stencil.opposite)[self.velocities]] + self.index] = 1
+        return no_stream_mask
+
+    # not sure about this. it probably makes sense to apply collisions here so that f[0] can adapt to the flow
+    #def make_no_collision_mask(self, f_shape):
+    #    no_collision_mask = torch.zeros(size=f_shape[1:], dtype=torch.bool, device=self.lattice.device)
+    #    no_collision_mask[self.index] = 1
+    #    return no_collision_mask
```

Comment:
As far as I understand collisions should be applied here since the equation explicitly includes the post collision distributions

Reply:
I think it refers to the post-collision distributions coming from inside the domain. The other ones are only relevant for computing u_w. I don't think the Krüger book is specific about how to compute it.


But to me it makes sense to collide here. I think the effect will be small anyway. But if you want to do some more digging, you are welcome.